common/persistence/cassandra/queue_v2_store.go

```diff
@@ -474,3 +476,32 @@ func (s *queueV2Store) getMaxMessageID(ctx context.Context, queueType persistenc
 	}
 	return maxMessageID, true, nil
 }
+
+func (s *queueV2Store) ListQueues(
+	ctx context.Context,
+	request *persistence.InternalListQueuesRequest,
+) (*persistence.InternalListQueuesResponse, error) {
+	if request.PageSize <= 0 {
+		return nil, persistence.ErrNonPositiveListQueuesPageSize
+	}
+	iter := s.session.Query(
+		templateGetQueueNamesQuery,
+		request.QueueType,
+	).PageSize(request.PageSize).PageState(request.NextPageToken).WithContext(ctx).Iter()
```

Comment:
Interesting--I didn't realize there was a generic pagination tool like this in Cassandra. Should the other queries be using it as well?

Reply:
Yeah, I just learned about it now. For ListQueues(), using this is the only option, since we don't have any IDs to do pagination. 

For ReadMessages(), we can either use ID or use Cassandra's pagination. Using ID-based pagination is good in this case because we can keep the logic the same for SQL and Cassandra.